crates/worker/src/processor/compile_processor.rs

```diff
@@ -0,0 +1,192 @@
+            tokio::spawn(async move {
+                // Clean compilation input files right away
+                if let Err(err) = s3_client.delete_dir(&dir).await {
+                    error!("Filed to delete compilation file: {err}")
+                }
+                if let Err(err) = sqs_client.delete_message(receipt_handle).await {
+                    error!("Failed to delete message from sqs: {err}");
+                }
+            });
+        }
+
+        match self.generate_presigned_urls(&file_keys).await {
+            Ok(presigned_urls) => {
+                self.purgatory
+                    .add_record(
+                        id,
+                        TaskResult::Success(TaskSuccess::Compile {
+                            presigned_urls: presigned_urls.clone(),
+                        }),
+                    )
+                    .await;
+                Ok(presigned_urls)
+            }
+            Err(err) => {
+                let task_result = TaskResult::Failure(TaskFailure {
+                    error_type: ServerError::InternalError,
+                    message: err.to_string(),
+                });
+                self.purgatory.add_record(id, task_result).await;
+
+                Err(err.into())
+            }
+        }
+    }
+
+    async fn upload_artifacts(
+        &self,
+        id: Uuid,
+        compilation_output: CompilationOutput,
+    ) -> anyhow::Result<Vec<String>> {
+        let auto_clean_up = AutoCleanUp {
+            dirs: vec![compilation_output.artifacts_dir.to_str().unwrap()],
+        };
+
+        let mut file_keys = Vec::with_capacity(compilation_output.artifacts_data.len());
+        for el in compilation_output.artifacts_data {
+            let absolute_path = compilation_output.artifacts_dir.join(&el.file_path);
+            let file_content = tokio::fs::File::open(absolute_path.clone())
+                .await
+                .map_err(anyhow::Error::from)
+                .with_context(|| format!("Couldn't open file: {}", absolute_path.display()))?;
+
+            let file_key = format!(
+                "{}/{}/{}",
+                ARTIFACTS_FOLDER,
+                id,
+                el.file_path.to_str().unwrap()
+            );
+            self.s3_client
+                .put_object(&file_key, file_content)
+                .await
+                .map_err(anyhow::Error::from)
+                .with_context(|| "Couldn't upload artifact")?; // TODO: TODO(101)
+            file_keys.push(file_key);
+        }
+
+        auto_clean_up.clean_up().await;
+        Ok(file_keys)
+    }
+
+    async fn generate_presigned_urls(&self, file_keys: &[String]) -> anyhow::Result<Vec<String>> {
+        const DOWNLOAD_URL_EXPIRATION: Duration = Duration::from_secs(5 * 60 * 60);
+
+        let mut presigned_urls = Vec::with_capacity(file_keys.len());
+        for el in file_keys {
+            let expires_in = PresigningConfig::expires_in(DOWNLOAD_URL_EXPIRATION).unwrap();
```

Comment:
It's just never fails with current implementation. If it does in case dependancies change, then we would fail all the time here anyway.

```rust
    pub fn build(self) -> Result<PresigningConfig, PresigningConfigError> {
        let expires_in = self.expires_in.ok_or(ErrorKind::ExpiresInRequired)?;
        if expires_in > ONE_WEEK {
            return Err(ErrorKind::ExpiresInDurationTooLong.into());
        }
        Ok(PresigningConfig {
            start_time: self.start_time.unwrap_or_else(
                // This usage is OK—customers can easily override this.
                #[allow(clippy::disallowed_methods)]
                SystemTime::now,
            ),
            expires_in,
        })
    }
```